### ISSUE TITLE
Bug: False positive with updated MD053 rule

### ISSUE THREAD

**Issue by rodcloutier** [igorshubovych/markdownlint-cli#316]
Version 0.32.2 introduces new fixes for MD053. It seems that with mkdocs, the given file gives a false positive.

```
$ cat test.md
# header

!!! notes

    The [link]

[link]: https://tekton.dev/docs/pipelines/pipelines/
$ markdownlint test.md
test.md:7:1 MD053/link-image-reference-definitions Link and image reference definitions should be needed [Unused link or image reference definition: "link"] [Context: "[link]: https://tekton.dev/do..."]
```

**Comment by DavidAnson** [comment]
Per the CommonMark specification, indenting text by four spaces makes it a code block and so the content within is not a link reference and the message is correct.

http://markdown-it.github.io/#md3=%7B%22source%22%3A%22%23%20header%5Cn%5Cn!!!%20notes%5Cn%5Cn%20%20%20%20The%20%5Blink%5D%5Cn%5Cn%5Blink%5D%3A%20https%3A%2F%2Ftekton.dev%2Fdocs%2Fpipelines%2Fpipelines%2F%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22html%22%7D%7D

**Comment by rodcloutier** [comment]
Thanks for pointing this out.  

I just found this to add support for admonitions https://github.com/DavidAnson/markdownlint/issues/207 which would probably be a way out for this.
